tests/main/sudo-env/task.yaml

```diff
@@ -0,0 +1,40 @@
+summary: verify that snap binaries dir is present in PATH under sudo
+
+description: |
+    Some distributions set secure_path in /etc/sudoers which resets the PATH
+    under sudo to some predefined set of locations. Make sure to account for all
+    distros supported by snapd that have sudo set up this way.
+
+environment:
+    # list of regular expressions that match systems where sudo is set up to use
+    # secure_path
+    SECURE_PATH_SUDO: "fedora-.* centos-.* amazon-linux-2-64 opensuse-.* debian-.*"
+
+debug: |
+    cat sudo.path || true
+    cat sudo-login.path || true
+
+execute: |
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB/dirs.sh"
+
+    # run a snap command via sudo
+    su -l -c "sudo sh -c 'echo :\$PATH:'" test > sudo.path
```

Comment:
Please use tests.session, su -l will give you a login shell and may trigger PAM.